plugins/vst_base/RemoteVstPlugin.cpp

```diff
@@ -402,6 +463,10 @@ class RemoteVstPlugin : public RemotePluginClient
 
 	bool m_processing;
 
+#ifdef NATIVE_LINUX_VST
+	pthread_mutex_t message_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Why not using `std::mutex`?

Reply:
If I understood correctly, this mutex is required because `m_messageList` is directly used for transferring messages from processing thread to UI thread. Is this correct?
@DomClark Compared to the `PostMessage` approach we're using on Windows, does this method have any (dis)advantages? If this way works better on Windows, can we consider having a consistent implementation on all platforms?

Reply:
> Why not using `std::mutex`?

I just used pthreads to create thread, so I used pthread everywhere. No special meaning. 

> If I understood correctly, this mutex is required because `m_messageList` is directly used for transferring messages from processing thread to UI thread. Is this correct?

Yes, `RemoteVstPlugin::processUIThreadMessages()` can be called from main thread (ui loop) and `queueMessage()` can be used in main or worker thread (`processingThread()`). `std::queue` is not thread save.

Reply:
This seems a reasonable alternative to `PostMessage` for now. Ideally we wouldn't use either as we aim for realtime safety - the pthreads version here has an explicit mutex, and `PostMessage` takes a lock internally. I plan to start working on this and other VST reworking/refactoring once this PR has been merged. Since we already have working Windows code, and this PR already has working Linux code, I don't think it's worth the effort to design a cross-platform lock-based system now, just to replace it with another design a little later.

Reply:
From my tests, `PostMessage` uses much less resources. Try adding out of 40 tracks with linux VST (the nanosleep message loop consumes the most of CPU time). I think in the future there should be only one thread for all VST plugins but that requires lots of changes in VST support (or better solution than nanosleep in guiEventLoop).